### PR TITLE
[ADR-183] OAuth2 client credentials token provider for cloud assets

### PR DESCRIPTION
**Pull request by jodavis-claude**
Jira: https://jodasoft.atlassian.net/browse/ADR-183

## What changed
- Added `ICloudAuthTokenProvider` interface with `GetTokenAsync(CancellationToken)` in `Services/CloudAssets/`
- Added `CognitoTokenProvider` singleton: POSTs to Cognito token endpoint with client credentials, caches token in-memory, proactively refreshes 60s before expiry, uses `SemaphoreSlim` for thread safety
- Added `ClientId`, `ClientSecret`, `CognitoTokenEndpointUrl` to `CloudSettings`
- Added log messages 1010–1012 to `MessageLogger.cs` for token acquisition events
- Added 5 unit tests covering first-call fetch, cached reuse, near-expiry refresh, cancellation, and HTTP failure propagation
- Updated `_doc_CloudAssets.md` with OAuth2 token provider section

## Test plan
- 396 app unit tests pass (includes new `CognitoTokenProviderTests`)
- 27 backend service tests pass
- 21 headless E2E tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_016PwpXFt999Q5Xk52qFukfu)_

### CODE REVIEW

**Review by jodavis-claude — COMMENTED** [review]
Automated review — see inline comments.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jodavis-claude** [comment]
**Re: Near-duplicate of CognitoTokenService**

The spec (ADR-183) explicitly calls for a new `ICloudAuthTokenProvider` interface and `CognitoTokenProvider` implementation in `Services/CloudAssets/`. `CognitoTokenService` lives in `Services/Backend/` and serves backend infrastructure with a different responsibility boundary. Consolidating them is a separate refactoring concern outside this task's scope.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jodavis-claude** [comment]
**Re: `new HttpClient()` anti-pattern**

This is consistent with the existing `CognitoTokenService` pattern already in the codebase. Migrating to `IHttpClientFactory` is a separate concern that would apply to both classes together as a future improvement, not something scoped to this task.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                Encoding.UTF8,
+                "application/json"),
+        };
+
+    // ─────────────────────────────────────────────────────────────────
+    // Tests
+    // ─────────────────────────────────────────────────────────────────
+
+    [TestMethod]
+    public async Task CognitoTokenProvider_GetTokenAsync_FirstCall_FetchesAndReturnsTokenAsync()
```
Comment (jodavis) [inline test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs]: Don't create async unit tests. Use TaskCompletionSource to simulate asynchronous operations, and have tests to cover all edge cases and error conditions (e.g. any call to an async method could block, throw, be cancelled, or succeed. All async calls that accept CancellationToken should respond to the CancellationToken passed in to `GetTokenAsync`).

File: test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            log.CognitoTokenProvider_AcquireTokenFailed(
+                new HttpRequestException("Response status code does not indicate success: 401 (Unauthorized)."));
+        });
+    }
+
+    // ─────────────────────────────────────────────────────────────────
+    // FakeHttpMessageHandler — returns pre-configured responses in order
+    // ─────────────────────────────────────────────────────────────────
+
+    private sealed class FakeHttpMessageHandler : HttpMessageHandler
```
Comment (jodavis) [inline test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs]: Move this to the TestUtilities namespace. It will probably be useful for other tests that need to make HTTP calls, e.g. the REST API calls that the real `CloudAssetDownloader` will eventually make.

File: test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        // Arrange
+        FakeHttpMessageHandler handler = new(BuildTokenResponse("first-token"));
+        using CognitoTokenProvider sut = MakeSut(httpMessageHandler: handler);
+
+        // Act
+        string token = await sut.GetTokenAsync(CancellationToken.None);
+
+        // Assert
+        token.Should().Be("first-token");
+        handler.CallCount.Should().Be(1);
```
Comment (jodavis) [inline test/AdaptiveRemote.App.Tests/Services/CloudAssets/CognitoTokenProviderTests.cs:54]: Should also verify that the correct URL/content was passed, not just that a call was made.

File: src/AdaptiveRemote.App/Services/CloudAssets/CognitoTokenProvider.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    private async Task<(string Token, DateTimeOffset Expiry)> AcquireTokenAsync(
+        CancellationToken ct)
+    {
+        List<KeyValuePair<string, string>> parameters =
+        [
+            new("grant_type", "client_credentials"),
+            new("client_id", _settings.ClientId),
+            new("client_secret", _settings.ClientSecret),
```
Comment (jodavis) [inline src/AdaptiveRemote.App/Services/CloudAssets/CognitoTokenProvider.cs:94]: If ClientId and ClientSecret are not set, this should fail gracefully. I'm not sure what that means for the API. Maybe it returns `null`, and the caller will know not to attach an `Authorization` header if the token is `null`?

File: src/AdaptiveRemote.App/Configuration/CloudAssetServiceExtensions.cs [hunk 5]
```diff
@@ -12,6 +12,7 @@ internal static class CloudAssetServiceExtensions
 {
     internal static IServiceCollection AddCloudAssetServices(this IServiceCollection services, IConfiguration configuration)
         => services
+            .AddSingleton<ICloudAuthTokenProvider, CognitoTokenProvider>()
```
Comment (jodavis) [inline src/AdaptiveRemote.App/Configuration/CloudAssetServiceExtensions.cs:15]: Nothing is calling this yet, so we have no idea whether the code actually works.

Let's add a call to `ICloudAuthTokenProvider.GetTokenAsync` in the stub implementation of `ICloudAssetDownloader`. It doesn't need a token, but it can try to acquire one anyway.

There should be E2E tests with ClientId/ClientSecret set and not set, so we validate that the application can start without credentials (no cloud downloads) and with credentials (cloud downloads attempted). I think we created a mock OAuth server to test authentication in ApiTests. These scenarios should be added to the CloudAsset E2E scenarios.

**Review by jodavis — CHANGES_REQUESTED** [review]

File: src/AdaptiveRemote.App/Configuration/CloudAssetServiceExtensions.cs (hunk 5 above)
Comment (jodavis) [reply src/AdaptiveRemote.App/Configuration/CloudAssetServiceExtensions.cs:15]: The `TestJwtAuthority` has been moved to `ISimulatedEnvironment` in the `feature/ADR-161-cusrtomization-service` branch, so it is available to all the client E2E tests as well. You should merge from that branch to get the updates, then you can write E2E tests for this service, including:
1. Valid credentials
2. Invalid credentials
3. No credentials

**Review by jodavis — CHANGES_REQUESTED** [review]

**Comment by jodavis** [comment]
@copilot Address the code review comments in this PR.

**Review by jodavis — APPROVED** [review]
